src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -114,3 +168,76 @@ def get_org_ids_with_scim(
                 flags=F("flags").bitor(AuthProvider.flags.scim_enabled)
             ).values_list("organization_id", flat=True)
         )
+
+
+class FakeRequestDict:
+    d: Mapping[str, str | bytes | None]
+
+    def __init__(self, **d: Any):
+        self.d = d
+
+    def __getitem__(self, item: str) -> str | bytes:
+        result = self.d[item]
+        if result is None:
+            raise KeyError(f"Key '{item}' does not exist")
+        return result
+
+    def __contains__(self, item: str) -> bool:
+        return self.d.get(item, None) is not None
+
+    def get(self, key: str, default: str | bytes | None = None) -> str | bytes | None:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+
+class FakeAuthenticationRequest:
```

Comment:
Why is this necessary?

Reply:
Also, the other advantage of this request implementation is that it controls the inputs and outputs.

For instance, it does not allow session updates (the access is immutable) or clears.  Any mutation or side effects to the session must be explicitly modeled and not implicit.

The mutatable attributes are the only two that are allowed to "result" from authentication: `auth` and `user`.

Reply:
In fact, as designed, any usage within the authentication implementation of a session or header variable that does not exist will fail the test.  *only explicit inputs and outputs*, no implicit sharing of the originating request object.